src/main/java/duke/storage/Storage.java

```diff
@@ -0,0 +1,159 @@
+     */
+    public TaskList load() throws UnableToParseException, IOException {
+        if (!Files.exists(path)) {
+            Files.createDirectories(path.getParent()); // create /data/ directory if it does not exist
+            Files.createFile(path);
+            return new TaskList();
+        }
+
+        List<String> lines = Files.readAllLines(this.path);
+        ArrayList<Task> tasks = new ArrayList<>();
+        for (String line : lines) {
+            String[] split = line.split(Pattern.quote(Storage.DELIMITER));
+            if (split.length == 0) {
+                throw new UnableToParseException(this.path.toString());
+            }
+
+            String description;
+            boolean isDone;
+            UnableToParseException e = new UnableToParseException("\"" + line + "\" at " + this.path);
+
+            switch (split[0]) {
+            case "T":
+                if (split.length != 3) {
+                    throw e;
+                }
+
+                isDone = Parser.parseStringToIsDone(split[1]);
+                description = split[2];
+                Todo todo = new Todo(description);
+                if (isDone) {
+                    todo.markAsDone();
+                }
+                tasks.add(todo);
+                break;
+            case "D":
+                if (split.length != 4) {
+                    throw e;
+                }
+
+                isDone = Parser.parseStringToIsDone(split[1]);
+                description = split[2];
+                LocalDate by;
+                try {
+                    by = LocalDate.parse(split[3]);
+                } catch (DateTimeParseException exception) {
+                    throw e;
+                }
+
+                Deadline deadline = new Deadline(description, by);
+                if (isDone) {
+                    deadline.markAsDone();
+                }
+                tasks.add(deadline);
+                break;
+            case "E":
+                if (split.length != 4) {
+                    throw e;
+                }
+
+                isDone = Parser.parseStringToIsDone(split[1]);
+                description = split[2];
+                String at = split[3];
+                Event event = new Event(description, at);
+                if (isDone) {
+                    event.markAsDone();
+                }
+                tasks.add(event);
+                break;
+            default:
+                throw e;
+            }
+        }
+        return new TaskList(tasks);
+    }
+
+    /**
+     * Solution adapted from https://stackoverflow.com/a/37624091 with modifications.
+     *
+     * @param index the line number to overwrite
+     * @param task the new task
+     * @throws IOException unexpected IO error.
+     */
+    public void update(int index, Task task) throws IOException {
+        List<String> fileContent = new ArrayList<>(Files.readAllLines(this.path));
```

Comment:
Actually, this seems a bit wasteful. Maybe you can just update/delete when the program exits, so you don't need to keep reading all lines and writing the entire task lists on a simple update/delete

Reply:
Hi, I think this has to been discussed in the forum [here](https://github.com/nus-cs2103-AY2122S1/forum/issues/62). The downside to that is that "The program might crash or be stopped by the user then the tasks wouldn't be saved"

Reply:
You can do a try catch all exception and save the file...though maybe this is not ideal.

Or you can add a shutdown hook
```Runtime.getRuntime().addShutdownHook(new Thread(() -> storeExistingTaskList()));```